packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  debug,
+  DebugSessionCustomEvent,
+  Disposable,
+  DebugSession as VscDebugSession,
+} from "vscode";
+
+export type DebugSessionDelegate = {
+  onConsoleLog(event: DebugSessionCustomEvent): void;
+  onDebuggerPaused(event: DebugSessionCustomEvent): void;
+  onDebuggerResumed(event: DebugSessionCustomEvent): void;
+};
+
+export class DebugSession implements Disposable {
+  private _session: VscDebugSession | undefined;
```

Comment:
we stopped using this pattern for private fields some time ago, would like to keep it off (and if you see it in other places also feel free to rename)